daemon/logger/logger.go

```diff
@@ -57,7 +58,7 @@ type LogWatcher struct {
 	Msg chan *Message
 	// For sending error messages that occur while while reading logs.
 	Err           chan error
-	closeNotifier chan struct{}
+	closeNotifier *pubsub.Publisher
```

Comment:
Also, we already used close-only-once pattern in code(should be in monitor), it includes select on this channel and close in default.